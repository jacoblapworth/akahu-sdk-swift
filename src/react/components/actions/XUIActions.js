import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-actions`;

const XUIActions = ({
  children,
  className,
  hasLayout,
  isLinear,
  primaryAction,
  qaHook,
  secondaryAction,
  tagName,
  ...spreadProps
}) => {
  const classes = cn(
    baseClass,
    className,
    hasLayout && `${baseClass}-layout`,
    isLinear && `${baseClass}-linear`,
  );
  const Tag = tagName;

  const clonedPrimary =
    primaryAction &&
    React.cloneElement(primaryAction, {
      className: cn(`${baseClass}--primary`, primaryAction.props.className),
    });

  const clonedSecondary =
    secondaryAction &&
    React.cloneElement(secondaryAction, {
      className: cn(`${baseClass}--secondary`, secondaryAction.props.className),
    });

  return (
    <Tag {...spreadProps} className={classes} data-automationid={qaHook}>
      {children}
      {clonedPrimary}
      {clonedSecondary}
    </Tag>
  );
};

export default XUIActions;

XUIActions.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * Applies standard layout settings. Defaults to true
   */
  hasLayout: PropTypes.bool,
  /**
   * Splits buttons to left and right for a linear flow. Can be combined with hasLayout
   */
  isLinear: PropTypes.bool,
  /**
   * XUIButton to style as primary. Alternately, pass in all the children
   */
  primaryAction: PropTypes.element,
  qaHook: PropTypes.string,
  /**
   * XUIButton to style as secondary
   */
  secondaryAction: PropTypes.element,
  /**
   * Wrapper element tag type, for semantic purposes (eg. panel footers). Defaults to "div"
   */
  tagName: PropTypes.string,
};

XUIActions.defaultProps = {
  hasLayout: true,
  tagName: 'div',
};
