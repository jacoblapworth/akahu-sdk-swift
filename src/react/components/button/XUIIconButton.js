import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import XUIButton from './XUIButton';
import XUIIcon from '../icon/XUIIcon';
import {
  sizeClassNames,
  iconSizeClassNames,
  variantClassNames,
  buttonTypes,
} from './private/constants';
import { colorClasses, wrapperSizeClasses, rotationClasses } from '../icon/private/constants';

export default class XUIIconButton extends PureComponent {
  render() {
    const {
      className,
      icon,
      ariaLabel,
      size,
      isInverted,
      iconSize,
      desc,
      role,
      rotation,
      iconColor,
      ...otherProps
    } = this.props;

    const iconSizeClass = iconSizeClassNames[size || 'medium'];

    return (
      <XUIButton
        ref={n => (this.rootNode = n && n.rootNode)}
        variant="unstyled"
        className={cn(
          className,
          iconSizeClass,
          variantClassNames[isInverted ? 'icon-inverted' : 'icon'],
        )}
        aria-label={ariaLabel}
        {...otherProps}
      >
        <XUIIcon
          icon={icon}
          size={iconSize}
          desc={desc}
          role={role}
          rotation={rotation}
          color={iconColor}
        />
      </XUIButton>
    );
  }
}

XUIIconButton.propTypes = {
  className: PropTypes.string,

  /** Required prop, an object describing the path, width and height. */
  icon: PropTypes.shape({
    path: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,

  /** Required prop, providing the label to any assistive technologies */
  ariaLabel: PropTypes.string.isRequired,

  /** Determines if the iconButton is disabled or not. */
  isDisabled: PropTypes.bool,

  /** If true, sets appropriate `rel` values to prevent new page from having access to
   * `window.opener`. Should be used for links pointing at external sites. * */
  isExternalLink: PropTypes.bool,

  /** A keydown event handler for the iconButton */
  onKeyDown: PropTypes.func,

  /** Bind a function to fire when the iconButton is clicked */
  onClick: PropTypes.func,

  /**
   * Modifier for the size of the iconButton. `medium`, `small`, or `xsmall`.
   */
  size: PropTypes.oneOf(Object.keys(sizeClassNames)),

  /** Whether or not to render this iconButton using an <a> tag */
  isLink: PropTypes.bool,

  /** The type attribute of this iconButton. `submit`, `button`, or `reset`. */
  type: PropTypes.oneOf(Object.keys(buttonTypes).map(type => buttonTypes[type])),

  /** The `href` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
  href: PropTypes.string,

  /** The `rel` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
  rel: PropTypes.string,

  /** The HTML tabIndex attribute value */
  tabIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),

  /** The `target` attribute to use on the anchor element (ignored unless `isLink` is `true`) */
  target: PropTypes.string,

  /** The `title` attribute */
  title: PropTypes.string,

  /** Adds a size modifier to the icon */
  iconSize: PropTypes.oneOf(Object.keys(wrapperSizeClasses)),

  /** Description of the icon to be read by screen readers */
  desc: PropTypes.string,

  /** Role to be applied to the icon SVG for screen readers */
  role: PropTypes.string,

  /** Adds a rotation modifier to the icon. Accepted values are 0 (default), 90, 180, 270 */
  rotation: PropTypes.oneOf([
    ...Object.keys(rotationClasses),
    ...Object.keys(rotationClasses).map(n => parseInt(n)),
  ]),

  /** Adds a color modifier to the icon */
  iconColor: PropTypes.oneOf(Object.keys(colorClasses)),

  /** Applies inverted styling */
  isInverted: PropTypes.bool,

  qaHook: PropTypes.string,
};

XUIIconButton.defaultProps = {
  tabIndex: 0,
  type: buttonTypes.button,
  isLink: false,
  isDisabled: false,
  isExternalLink: false,
};
