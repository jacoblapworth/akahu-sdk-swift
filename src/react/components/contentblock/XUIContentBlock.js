import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-contentblock`;

const XUIContentBlock = ({ children, className, qaHook }) => {
  const listClasses = cn(className, baseClass);

  return (
    <div className={listClasses} data-automationid={qaHook}>
      {children}
    </div>
  );
};

export default XUIContentBlock;

XUIContentBlock.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  qaHook: PropTypes.string,
};
