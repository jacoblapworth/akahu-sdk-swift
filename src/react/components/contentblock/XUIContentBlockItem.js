import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';
import { isKeyClick } from '../helpers/reactKeyHandler';
import shouldRender from '../helpers/shouldRender';

const XUIContentBlockItem = ({
  _ariaControls,
  _ariaExpanded,
  _isAccordionTrigger,
  action,
  children,
  className,
  description,
  hasBottomRadius,
  hasLayout,
  hasTopRadius,
  href,
  isRowLink,
  leftContent,
  onClick,
  onKeyDown,
  overflow,
  pinnedValue,
  primaryHeading,
  qaHook,
  secondaryHeading,
  tagPosition,
  tags,
}) => {
  const baseClass = _isAccordionTrigger ? `${ns}-accordiontrigger` : `${ns}-contentblockitem`;

  const clonedAction =
    action &&
    React.cloneElement(action, {
      className: cn(action.props.className, `${baseClass}--actions`),
    });

  const tagPositionInline = tagPosition === 'inline';

  const builtHeadings = shouldRender(primaryHeading) && (
    <div className={`${baseClass}--headings`}>
      <span className={`${baseClass}--primaryheading`}>{primaryHeading}</span>
      {shouldRender(secondaryHeading) && (
        <span className={`${baseClass}--secondaryheading`}>{secondaryHeading}</span>
      )}
      {tagPositionInline && tags}
    </div>
  );

  const tagPositionDescription = tagPosition === 'description';

  let builtDescriptionArea = description && (
    <div className={`${baseClass}--description`}>
      <span className={`${baseClass}--description--text`}>{description}</span>
      {tagPositionDescription && tags}
    </div>
  );

  if (!description && tags && tagPositionDescription) {
    builtDescriptionArea = <div className={`${baseClass}--description`}>{tags}</div>;
  }

  const builtPinnedValue = pinnedValue && (
    <div className={`${baseClass}--pinnedvalue`}>{pinnedValue}</div>
  );

  const leftContentClasses = cn(
    `${baseClass}--leftcontent`,
    description && `${baseClass}--leftcontent-layout`,
  );

  const builtLeftContent = leftContent && <div className={leftContentClasses}>{leftContent}</div>;

  const builtMainContent = (builtHeadings || builtDescriptionArea) && (
    <div className={`${baseClass}--maincontent`}>
      {builtHeadings}
      {builtDescriptionArea}
    </div>
  );

  const tagPositionRight = tagPosition === 'right';

  const builtRightContent = (builtPinnedValue ||
    action ||
    overflow ||
    (tags && tagPositionRight)) && (
    <div
      className={`${baseClass}--rightcontent`}
      onClick={event => {
        href && event.preventDefault();
        event.stopPropagation();
      }}
      onKeyDown={event => {
        if (isKeyClick(event)) {
          href && event.preventDefault();
          event.stopPropagation();
        }
      }}
      role="presentation"
    >
      {tagPositionRight && tags}
      {builtPinnedValue}
      {clonedAction}
      {overflow}
    </div>
  );

  const divClasses = cn(
    `${baseClass}`,
    hasLayout && `${baseClass}-layout`,
    isRowLink && `${baseClass}-rowlink`,
    hasTopRadius && `${baseClass}-has-top-radius`,
    hasBottomRadius && `${baseClass}-has-bottom-radius`,
    className,
  );

  const Tag = href ? 'a' : 'div';
  const role = (onClick || onKeyDown) && !href ? 'button' : undefined;

  return (
    <Tag
      aria-controls={_ariaControls}
      aria-expanded={_ariaExpanded}
      className={divClasses}
      data-automationid={qaHook}
      href={href}
      onClick={onClick}
      onKeyDown={onKeyDown}
      role={role}
      tabIndex={onClick || onKeyDown || href ? 0 : undefined}
    >
      {builtLeftContent}
      {children}
      {builtMainContent}
      {builtRightContent}
    </Tag>
  );
};

export default XUIContentBlockItem;

XUIContentBlockItem.propTypes = {
  /**
   * For accordion: identifies content associated with the trigger
   * @ignore
   */
  _ariaControls: PropTypes.string,

  /**
   * For accordion: denotes whether content associated with the trigger is expanded
   * @ignore
   */
  _ariaExpanded: PropTypes.bool,

  /**
   * Gives accordion items the proper name for styling and semantics
   * @ignore
   */
  _isAccordionTrigger: PropTypes.bool,

  /**
   * Optional actions to be right aligned. Use the XUIActions component.
   */
  action: PropTypes.element,

  children: PropTypes.node,
  className: PropTypes.string,

  /**
   * Description to be placed under the heading
   */
  description: PropTypes.node,

  /**
   * Determines whether to apply bottom left and bottom right border radius on the content block item
   */
  hasBottomRadius: PropTypes.bool,

  /**
   * Determines whether to apply default layout styling or not
   */
  hasLayout: PropTypes.bool,

  /**
   * Determines whether to apply top left and top right border radius on the content block item
   */
  hasTopRadius: PropTypes.bool,

  /**
   * The `href` attribute to use on the anchor element
   */
  href: PropTypes.string,

  /**
   * Determines whether to apply hover styling on the entire content block item
   */
  isRowLink: PropTypes.bool,

  /**
   * Left most component option, typically an `avatar`, `checkbox` or `rollover checkbox` component
   */
  leftContent: PropTypes.element,

  /**
   * Callback to fire when content block item is clicked.
   */
  onClick: PropTypes.func,

  /**
   * Callback to fire on keyDown for each content block item
   */
  onKeyDown: PropTypes.func,

  /**
   * Any component passed as right most content, typically a `dropdown toggled` component
   */
  overflow: PropTypes.element,

  /**
   * Text pinned to right side of content block
   */
  pinnedValue: PropTypes.node,

  /**
   * Primary heading for the content block
   */
  primaryHeading: PropTypes.node,

  qaHook: PropTypes.string,

  /**
   * Secondary heading for the content block
   */
  secondaryHeading: PropTypes.node,

  /**
   * Repositions the tags in other places around the component
   */
  tagPosition: PropTypes.oneOf(['description', 'right', 'inline']),

  /**
   * Tag or other user determined node to go to right of primary heading
   */
  tags: PropTypes.oneOfType([PropTypes.element, PropTypes.array]),
};

XUIContentBlockItem.defaultProps = {
  _isAccordionTrigger: false,
  hasLayout: true,
  tagPosition: 'description',
};
