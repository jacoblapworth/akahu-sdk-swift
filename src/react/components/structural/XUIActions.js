import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-actions`;

export default class XUIActions extends PureComponent {
	render() {
		const {
			className,
			children,
			hasLayout,
			isLinear,
			primaryAction,
			secondaryAction,
			tagName,
			...spreadProps
		} = this.props;
		const classes = cn(
			baseClass,
			className,
			hasLayout && `${baseClass}-layout`,
			isLinear && `${baseClass}-linear`,
		);
		const Tag = tagName;

		const clonedPrimary = primaryAction && React.cloneElement(primaryAction, {
			className: cn(`${baseClass}--primary`, {
				[`${baseClass}--button-spacing`]: hasLayout && !isLinear,
			}),
		});

		const clonedSecondary = secondaryAction && React.cloneElement(secondaryAction, {
			className: `${baseClass}--secondary`,
		});

		return (
			<Tag {...spreadProps} className={classes}>
				{children}
				{clonedPrimary}
				{clonedSecondary}
			</Tag>
		);
	}
}

XUIActions.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	/**
	 * Applies standard layout settings. Defaults to true
	 */
	hasLayout: PropTypes.bool,
	/**
	 * Splits buttons to left and right for a linear flow. Can be combined with hasLayout
	 */
	isLinear: PropTypes.bool,
	/**
	 * XUIButton to style as primary. Alternately, pass in all the children
	 */
	primaryAction: PropTypes.element,
	/**
	 * XUIButton to style as secondary
	 */
	secondaryAction: PropTypes.element,
	/**
	 * Wrapper element tag type, for semantic purposes (eg. panel footers). Defaults to "div"
	 */
	tagName: PropTypes.string,
};

XUIActions.defaultProps = {
	hasLayout: true,
	tagName: 'div',
};
