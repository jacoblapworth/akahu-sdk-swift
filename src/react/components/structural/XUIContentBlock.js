import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-contentblock`;

export default class XUIContentBlock extends PureComponent {
	render() {
		const {
			className,
			children,
		} = this.props;
		const listClasses = cn(className, baseClass);

		return (
			<ol className={listClasses}>
				{children}
			</ol>
		);
	}
}

XUIContentBlock.propTypes = {
	className: PropTypes.string,
	children: PropTypes.node,
};

XUIContentBlock.defaultProps = {};
