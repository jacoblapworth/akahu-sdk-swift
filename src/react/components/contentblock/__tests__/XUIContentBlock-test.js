import React from 'react';
import XUIButton from '../../button/XUIButton';
import XUIIconButton from '../../button/XUIIconButton';
import XUIActions from '../../actions/XUIActions';
import XUIContentBlock from '../XUIContentBlock';
import XUIContentBlockItem from '../XUIContentBlockItem';
import overflow from '@xero/xui-icon/icons/overflow';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('<XUI ContentBlock and ContentBlockItem/>', () => {
  const qaHook = 'qaHook';
  const testPrimaryHeading = 'test primary heading';

  it('renders the base content block with no children', () => {
    const testContentblock = renderer.create(<XUIContentBlock />);
    expect(testContentblock).toMatchSnapshot();
  });
  it('renders content block with content block item child and headings', () => {
    const testContentblockWithChild = renderer.create(
      <XUIContentBlock>
        <XUIContentBlockItem primaryHeading={testPrimaryHeading} />
      </XUIContentBlock>,
    );
    expect(testContentblockWithChild).toMatchSnapshot();
  });
  it('renders extra content block item classes that are passed in', () => {
    const tag = shallow(<XUIContentBlockItem className="testClass" />);
    expect(tag.hasClass('testClass')).toEqual(true);
  });
  it('renders content block item without default layout', () => {
    const wrapper = shallow(<XUIContentBlockItem hasLayout={false} />);
    expect(wrapper.hasClass('xui-contentblockitem-layout')).toEqual(false);
  });
  it('renders content block item with everything', () => {
    const testOverflow = <XUIIconButton icon={overflow} ariaLabel="Overflow Menu" />;
    const testLeftContent = (
      <abbr className="xui-avatar xui-avatar-color-2" role="presentation">
        P
      </abbr>
    );
    const testActions = (
      <XUIActions
        primaryAction={
          <XUIButton key="one" variant="primary" size="small">
            One
          </XUIButton>
        }
        secondaryAction={
          <XUIButton key="two" size="small">
            Two
          </XUIButton>
        }
      />
    );
    const testTag = (
      <span className="xui-tag xui-tag-positive xui-margin-left-small">Positive</span>
    );

    const testContentblockWithEverything = renderer.create(
      <XUIContentBlock>
        <XUIContentBlockItem
          isRowLink
          hasTopRadius
          hasBottomRadius
          primaryHeading={testPrimaryHeading}
          secondaryHeading="test secondary heading"
          overflow={testOverflow}
          leftContent={testLeftContent}
          actions={testActions}
          pinnedValue="0.00"
          tags={testTag}
        />
      </XUIContentBlock>,
    );
    expect(testContentblockWithEverything).toMatchSnapshot();
  });

  it('renders content block item tag position description by default', () => {
    const testOverflow = <XUIIconButton icon={overflow} ariaLabel="Overflow Menu" />;
    const testLeftContent = (
      <abbr className="xui-avatar xui-avatar-color-2" role="presentation">
        P
      </abbr>
    );
    const testActions = (
      <XUIActions
        primaryAction={
          <XUIButton key="one" variant="primary" size="small">
            One
          </XUIButton>
        }
        secondaryAction={
          <XUIButton key="two" size="small">
            Two
          </XUIButton>
        }
      />
    );
    const testTag = (
      <span className="xui-tag xui-tag-positive xui-margin-left-small">Positive</span>
    );

    const testContentblockWithEverything = renderer.create(
      <XUIContentBlock>
        <XUIContentBlockItem
          isRowLink
          hasTopRadius
          hasBottomRadius
          primaryHeading={testPrimaryHeading}
          secondaryHeading="test secondary heading"
          overflow={testOverflow}
          leftContent={testLeftContent}
          actions={testActions}
          pinnedValue="0.00"
          tags={testTag}
        />
      </XUIContentBlock>,
    );
    expect(testContentblockWithEverything).toMatchSnapshot();
  });

  it('renders content block item tag position inline', () => {
    const testOverflow = <XUIIconButton icon={overflow} ariaLabel="Overflow Menu" />;
    const testLeftContent = (
      <abbr className="xui-avatar xui-avatar-color-2" role="presentation">
        P
      </abbr>
    );
    const testActions = (
      <XUIActions
        primaryAction={
          <XUIButton key="one" variant="primary" size="small">
            One
          </XUIButton>
        }
        secondaryAction={
          <XUIButton key="two" size="small">
            Two
          </XUIButton>
        }
      />
    );
    const testTag = (
      <span className="xui-tag xui-tag-positive xui-margin-left-small">Positive</span>
    );

    const testContentblockWithEverything = renderer.create(
      <XUIContentBlock>
        <XUIContentBlockItem
          isRowLink
          hasTopRadius
          hasBottomRadius
          primaryHeading={testPrimaryHeading}
          secondaryHeading="test secondary heading"
          overflow={testOverflow}
          leftContent={testLeftContent}
          actions={testActions}
          pinnedValue="0.00"
          tags={testTag}
          tagPosition="inline"
        />
      </XUIContentBlock>,
    );
    expect(testContentblockWithEverything).toMatchSnapshot();
  });

  it('renders content block item tag position right', () => {
    const testOverflow = <XUIIconButton icon={overflow} ariaLabel="Overflow Menu" />;
    const testLeftContent = (
      <abbr className="xui-avatar xui-avatar-color-2" role="presentation">
        P
      </abbr>
    );
    const testActions = (
      <XUIActions
        primaryAction={
          <XUIButton key="one" variant="primary" size="small">
            One
          </XUIButton>
        }
        secondaryAction={
          <XUIButton key="two" size="small">
            Two
          </XUIButton>
        }
      />
    );
    const testTag = (
      <span className="xui-tag xui-tag-positive xui-margin-left-small">Positive</span>
    );
    const testHrefProp = '#';

    const testContentblockWithEverything = renderer.create(
      <XUIContentBlock>
        <XUIContentBlockItem
          isRowLink
          hasTopRadius
          hasBottomRadius
          primaryHeading={testPrimaryHeading}
          secondaryHeading="test secondary heading"
          overflow={testOverflow}
          leftContent={testLeftContent}
          actions={testActions}
          pinnedValue="0.00"
          tags={testTag}
          tagPosition="right"
          tag={testTag}
          href={testHrefProp}
        />
      </XUIContentBlock>,
    );
    expect(testContentblockWithEverything).toMatchSnapshot();
  });
  it('renders content block item main content as anchor when onKeyDown prop is passed', () => {
    const testOnKeyDownCallback = () => {};
    const wrapper = renderer.create(<XUIContentBlockItem onKeyDown={testOnKeyDownCallback} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('invokes callback passed to onClick prop', () => {
    const testCallback = jest.fn();
    const wrapper = shallow(
      <XUIContentBlockItem primaryHeading={testPrimaryHeading} onClick={testCallback} />,
    );
    wrapper.find('[role="button"]').simulate('click');
    expect(testCallback).toHaveBeenCalledTimes(1);
  });
  it('renders content block and content block item with automation id when qaHook prop is passed in', () => {
    const wrapper = renderer.create(
      <XUIContentBlock qaHook={qaHook}>
        <XUIContentBlockItem qaHook={qaHook} />
      </XUIContentBlock>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
