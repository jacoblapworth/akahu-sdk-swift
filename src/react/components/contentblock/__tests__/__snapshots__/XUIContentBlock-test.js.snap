// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block and content block item with automation id when qaHook prop is passed in 1`] = `
<div
  className="xui-contentblock"
  data-automationid="qaHook"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout"
    data-automationid="qaHook"
  />
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block item main content as anchor when onKeyDown prop is passed 1`] = `
<div
  className="xui-contentblockitem xui-contentblockitem-layout"
/>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block item tag position description by default 1`] = `
<div
  className="xui-contentblock"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout xui-contentblockitem-rowlink xui-contentblockitem-has-top-radius xui-contentblockitem-has-bottom-radius"
  >
    <div
      className="xui-contentblockitem--leftcontent"
    >
      <abbr
        className="xui-avatar xui-avatar-color-2"
        role="presentation"
      >
        P
      </abbr>
    </div>
    <div
      className="xui-contentblockitem--maincontent"
    >
      <div
        className="xui-contentblockitem--headings"
      >
        <span
          className="xui-contentblockitem--primaryheading"
        >
          test primary heading
        </span>
        <span
          className="xui-contentblockitem--secondaryheading"
        >
          test secondary heading
        </span>
      </div>
      <div
        className="xui-contentblockitem--description"
      >
        <span
          className="xui-tag xui-tag-positive xui-margin-left-small"
        >
          Positive
        </span>
      </div>
    </div>
    <div
      className="xui-contentblockitem--rightcontent"
    >
      <div
        className="xui-contentblockitem--pinnedvalue"
      >
        0.00
      </div>
      <button
        aria-label="Overflow Menu"
        className="xui-button xui-button-icon-medium xui-button-icon"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          className="xui-icon"
          focusable="false"
          height={13}
          viewBox="0 0 3 13"
          width={3}
        >
          <path
            d="M1.5 3a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0-5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            role="presentation"
          />
        </svg>
        <div
          className="xui-touchtarget"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block item tag position inline 1`] = `
<div
  className="xui-contentblock"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout xui-contentblockitem-rowlink xui-contentblockitem-has-top-radius xui-contentblockitem-has-bottom-radius"
  >
    <div
      className="xui-contentblockitem--leftcontent"
    >
      <abbr
        className="xui-avatar xui-avatar-color-2"
        role="presentation"
      >
        P
      </abbr>
    </div>
    <div
      className="xui-contentblockitem--maincontent"
    >
      <div
        className="xui-contentblockitem--headings"
      >
        <span
          className="xui-contentblockitem--primaryheading"
        >
          test primary heading
        </span>
        <span
          className="xui-contentblockitem--secondaryheading"
        >
          test secondary heading
        </span>
        <span
          className="xui-tag xui-tag-positive xui-margin-left-small"
        >
          Positive
        </span>
      </div>
    </div>
    <div
      className="xui-contentblockitem--rightcontent"
    >
      <div
        className="xui-contentblockitem--pinnedvalue"
      >
        0.00
      </div>
      <button
        aria-label="Overflow Menu"
        className="xui-button xui-button-icon-medium xui-button-icon"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          className="xui-icon"
          focusable="false"
          height={13}
          viewBox="0 0 3 13"
          width={3}
        >
          <path
            d="M1.5 3a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0-5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            role="presentation"
          />
        </svg>
        <div
          className="xui-touchtarget"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block item tag position right 1`] = `
<div
  className="xui-contentblock"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout xui-contentblockitem-rowlink xui-contentblockitem-has-top-radius xui-contentblockitem-has-bottom-radius"
  >
    <div
      className="xui-contentblockitem--leftcontent"
    >
      <abbr
        className="xui-avatar xui-avatar-color-2"
        role="presentation"
      >
        P
      </abbr>
    </div>
    <a
      className="xui-contentblockitem--maincontent"
      href="#"
    >
      <div
        className="xui-contentblockitem--headings"
      >
        <span
          className="xui-contentblockitem--primaryheading"
        >
          test primary heading
        </span>
        <span
          className="xui-contentblockitem--secondaryheading"
        >
          test secondary heading
        </span>
      </div>
    </a>
    <div
      className="xui-contentblockitem--rightcontent"
    >
      <span
        className="xui-tag xui-tag-positive xui-margin-left-small"
      >
        Positive
      </span>
      <div
        className="xui-contentblockitem--pinnedvalue"
      >
        0.00
      </div>
      <button
        aria-label="Overflow Menu"
        className="xui-button xui-button-icon-medium xui-button-icon"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          className="xui-icon"
          focusable="false"
          height={13}
          viewBox="0 0 3 13"
          width={3}
        >
          <path
            d="M1.5 3a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0-5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            role="presentation"
          />
        </svg>
        <div
          className="xui-touchtarget"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block item with everything 1`] = `
<div
  className="xui-contentblock"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout xui-contentblockitem-rowlink xui-contentblockitem-has-top-radius xui-contentblockitem-has-bottom-radius"
  >
    <div
      className="xui-contentblockitem--leftcontent"
    >
      <abbr
        className="xui-avatar xui-avatar-color-2"
        role="presentation"
      >
        P
      </abbr>
    </div>
    <div
      className="xui-contentblockitem--maincontent"
    >
      <div
        className="xui-contentblockitem--headings"
      >
        <span
          className="xui-contentblockitem--primaryheading"
        >
          test primary heading
        </span>
        <span
          className="xui-contentblockitem--secondaryheading"
        >
          test secondary heading
        </span>
      </div>
      <div
        className="xui-contentblockitem--description"
      >
        <span
          className="xui-tag xui-tag-positive xui-margin-left-small"
        >
          Positive
        </span>
      </div>
    </div>
    <div
      className="xui-contentblockitem--rightcontent"
    >
      <div
        className="xui-contentblockitem--pinnedvalue"
      >
        0.00
      </div>
      <button
        aria-label="Overflow Menu"
        className="xui-button xui-button-icon-medium xui-button-icon"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          className="xui-icon"
          focusable="false"
          height={13}
          viewBox="0 0 3 13"
          width={3}
        >
          <path
            d="M1.5 3a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 10a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0-5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            role="presentation"
          />
        </svg>
        <div
          className="xui-touchtarget"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders content block with content block item child and headings 1`] = `
<div
  className="xui-contentblock"
>
  <div
    className="xui-contentblockitem xui-contentblockitem-layout"
  >
    <div
      className="xui-contentblockitem--maincontent"
    >
      <div
        className="xui-contentblockitem--headings"
      >
        <span
          className="xui-contentblockitem--primaryheading"
        >
          test primary heading
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<XUI ContentBlock and ContentBlockItem/> renders the base content block with no children 1`] = `
<div
  className="xui-contentblock"
/>
`;
